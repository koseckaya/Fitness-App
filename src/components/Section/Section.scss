

.section {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 20px 0;

  &__title {
    font-size: 18px;
  }
  &__header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 24px;
  }
  &__content{
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    row-gap: 20px;
  }
}
