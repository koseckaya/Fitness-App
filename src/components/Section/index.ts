export { default as Section } from './Section';
export type { Props as SectionProps } from './Section';
