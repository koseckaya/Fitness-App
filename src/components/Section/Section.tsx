import { FC } from 'react';
import './Section.scss';

export type Props = {
    className?: string;
   
};

const Section: FC<Props> = ({ className}: Props) => (
    <div
        className={`section ${className}`}
    >
        <span>{className}</span>
    </div>
);

export default Section;
