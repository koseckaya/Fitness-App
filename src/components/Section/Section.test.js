import React from 'react';
import renderer from 'react-test-renderer';
import Section from './Section';

const props = {
    title: "Free Workout Programs",
    btnText: "Free Workout Programs",
    children: `<div></div>`,
    onClick: () => { },
}



describe("Section component", () => {
    it('should render component with default props', () => {
        const tree = renderer
            .create(<Section {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render component without title props', () => {
        const newProps = { ...props, title: '' }
        const component = renderer
            .create(<Section {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without btnText props', () => {
        const newProps = { ...props, btnText: '' }
        const component = renderer
            .create(<Section {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without children props', () => {
        const newProps = { ...props, children: '' }
        const component = renderer
            .create(<Section {...newProps} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})