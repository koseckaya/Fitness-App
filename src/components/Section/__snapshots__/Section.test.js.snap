// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should render component with default props 1`] = `
<div
  className="container "
>
  <div
    className="section"
  >
    <div
      className="section__header"
    >
      <div
        className="section__title"
      >
        Free Workout Programs
      </div>
      <div
        className="button  button-primary"
        onClick={[Function]}
      >
        Free Workout Programs
      </div>
    </div>
    <div
      className="section__content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;

exports[`Section component should render component without btnText props 1`] = `
<div
  className="container "
>
  <div
    className="section"
  >
    <div
      className="section__header"
    >
      <div
        className="section__title"
      >
        Free Workout Programs
      </div>
      <div
        className="button  button-primary"
        onClick={[Function]}
      />
    </div>
    <div
      className="section__content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;

exports[`Section component should render component without children props 1`] = `
<div
  className="container "
>
  <div
    className="section"
  >
    <div
      className="section__header"
    >
      <div
        className="section__title"
      >
        Free Workout Programs
      </div>
      <div
        className="button  button-primary"
        onClick={[Function]}
      >
        Free Workout Programs
      </div>
    </div>
    <div
      className="section__content"
    />
  </div>
</div>
`;

exports[`Section component should render component without title props 1`] = `
<div
  className="container "
>
  <div
    className="section"
  >
    <div
      className="section__header"
    >
      <div
        className="section__title"
      />
      <div
        className="button  button-primary"
        onClick={[Function]}
      >
        Free Workout Programs
      </div>
    </div>
    <div
      className="section__content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;
